test/integration/integration_test.cc

```diff
@@ -521,6 +521,14 @@ TEST_P(IntegrationTest, TestSmuggling) {
     sendRawHttpAndWaitForResponse(lookupPort("http"), full_request.c_str(), &response, false);
     EXPECT_THAT(response, StartsWith("HTTP/1.1 400 Bad Request\r\n"));
   }
+  {
+    std::string response;
+    const std::string full_request = "GET / HTTP/1.1\r\nHost: host\r\ncontent-length: "
+                                     "36\r\n\ttransfer-encoding: chunked\r\n\r\n" +
```

Comment:
Would it be possible to break this request string at \r\n boundaries?

Something like:
```
"GET / HTTP/1.1\r\n"
"Host: host\r\n"
"content-length: 36\r\n"
"\ttransfer-encoding: chunked\r\n\r\n"
```

It took me a while to figure out that the difference between this request and the one above is the \t just before transfer-encoding.  A comment about this being about \t would be helpful.

Also, consider a test of transfer-encoding with \t before it but no content-length, and \t before content-length and no transfer-encoding header.